GA Prompt Node Usage and Enable/Disable all mocks toggle (#324)
vellum-ai/vellum-client-generator: fern/docs/content/change-log/2024-07.mdx

title: Changelog | July, 2024
---

## Prompt Node Usage in Workflows

_July 25th, 2024_

> You can now see token counts and other usage statistics appear in Prompt Node results when invoking Workflows in the Workflow Sandbox:

Reviewer: ```suggestion
You can now see token counts and other usage metrics appear in Prompt Node results when invoking Workflows in the Workflow Sandbox:
```

> ![Prompt Node Usage](https://storage.cloud.google.com/vellum-public/help-docs/changelogs/2024-07/prompt-node-usage.png)

Reviewer: Let's please replace this screenshot with this one:

<img width="502" alt="2024-07-25_09-37-13" src="https://github.com/user-attachments/assets/0c026009-5765-40b9-9535-20d61b973a04">

> This setting is now on by default, but could be toggled off in the Workflow Builder Settings. This setting is also now available for _Workflow Deployments_, by passing in `True`

Reviewer: ```suggestion
This setting is now on by default, but can be toggled off in the Workflow Builder Settings.

You can also now return usage data when invoking a _Workflow Deployment_ via API, by passing in `True`
```

to the `expand_meta.usage` parameter on either Execute Workflow endpoints.

```python
stream = client.execute_workflow_stream(
  workflow_deployment_name="demo",
  inputs=[
    WorkflowRequestInputRequest_String(
      type="STRING",
      name="foo",
      value="bar",
    ),
  ],
  event_types=["WORKFLOW", "NODE"],
  expand_meta=WorkflowExpandMetaRequest(
    usage=True
  )
)

for event in stream:
  if event.type == "NODE" and event.data.state == "FULFILLED":
    node_result_data = event.data.data
    if node_result_data and node_result_data.type == "PROMPT":
      print(node_result_data.data.execution_meta.usage)
```

> ## Enable/Disable All Mocks toggle

Reviewer: ```suggestion
## Enable/Disable All Workflow Node Mocks
```

_July 25th, 2024_

Mocking Prompt Nodes helps to save token usage and time when developing the later stages of your Workflow. However, once the Workflow is in a good state, it's often useful to run

> the full Workflow end to end without mocks to make sure it all comes together. Previously, you had to enable/disable each mock individually. Now, beneath the scenario inputs there

Reviewer: ```suggestion
the full Workflow end-to-end without mocks to make sure it all comes together. Previously, you had to enable/disable each mock individually. Now, beneath the scenario inputs there
```

is a toggle that allows you to enable/disable all mocks in a workflow at once.

> ![Enable Disable All Mocks](https://storage.cloud.google.com/vellum-public/help-docs/changelogs/2024-07/enable-disable-all-mocks.png)

Reviewer: Let's please replace this screenshot with this one:

<img width="524" alt="2024-07-25_09-36-12" src="https://github.com/user-attachments/assets/addb630f-1e83-4516-ac6a-d0ae67026cf3">

Author: The difference being (in both) the Arrow overlay, correct?

Reviewer: 1. Showing more of the screen so that users can contextualize where this new UI lives; and
2. Using light mode instead of dark mode (generally preferred for new screenshots going forward in our docs)

The arrow is just a nice to have compared to the two above.
